mozetl/tab_spinner/utils.py

```diff
@@ -0,0 +1,148 @@
+    short_y = y[1]
+
+    if long_x is None:
+        long_x = pd.Series()
+    if long_y is None:
+        long_y = pd.Series()
+    if short_x is None:
+        short_x = pd.Series()
+    if short_y is None:
+        short_y = pd.Series()
+
+    return (long_x.add(long_y, fill_value=0.0), short_x.add(short_y, fill_value=0.0))
+
+def bucket_by_long_severity_per_client(spinner_pair):
+    buildId = spinner_pair[0][0]
+    hist = spinner_pair[1][0]
+    named_index = ["unaffected",
+                   "0ms - 999ms",
+                   "1000ms - 2296ms",
+                   "2297ms - 5276ms",
+                   "5277ms - 12123ms",
+                   "12124ms - 27855ms",
+                   "27856ms - 63999ms",
+                   "64000ms+"]
+
+    severity = pd.Series([0, 0, 0, 0, 0, 0, 0, 0], index=named_index)
+
+    if hist is None or hist.empty:
+        severity[named_index[0]] = 1
+    elif hist[hist.index >= 64000].sum() > 0:
+        severity[named_index[7]] = 1
+    elif hist[hist.index >= 27856].sum() > 0:
+        severity[named_index[6]] = 1
+    elif hist[hist.index >= 12124].sum() > 0:
+        severity[named_index[5]] = 1
+    elif hist[hist.index >= 5277].sum() > 0:
+        severity[named_index[4]] = 1
+    elif hist[hist.index >= 2297].sum() > 0:
+        severity[named_index[3]] = 1
+    elif hist[hist.index >= 1000].sum() > 0:
+        severity[named_index[2]] = 1
+    elif hist[hist.index >= 0].sum() > 0:
+        severity[named_index[1]] = 1
+
+    return (buildId, severity)
+
+def bucket_by_short_severity_per_client(spinner_pair):
+    buildId = spinner_pair[0][0]
+    long_hist = spinner_pair[1][0]
+    hist = spinner_pair[1][1]
+
+    named_index = ["unaffected",
+                   "not short",
+                   "0ms - 49ms",
+                   "50ms - 99ms",
+                   "100ms - 199ms",
+                   "200ms - 399ms",
+                   "400ms - 799ms",
+                   "800ms+"]
+
+    severity = pd.Series([0, 0, 0, 0, 0, 0, 0, 0], index=named_index)
+
+    if hist is None or hist.empty or long_hist is None or long_hist.empty:
+        severity[named_index[0]] = 1
+    elif long_hist[long_hist.index >= 1000].sum() > 0:
+        severity[named_index[1]] = 1
+    elif hist[hist.index >= 800].sum() > 0:
+        severity[named_index[7]] = 1
+    elif hist[hist.index >= 400].sum() > 0:
+        severity[named_index[6]] = 1
+    elif hist[hist.index >= 200].sum() > 0:
+        severity[named_index[5]] = 1
+    elif hist[hist.index >= 100].sum() > 0:
+        severity[named_index[4]] = 1
+    elif hist[hist.index >= 50].sum() > 0:
+        severity[named_index[3]] = 1
+    elif hist[hist.index >= 0].sum() > 0:
+        severity[named_index[2]] = 1
+
+    return (buildId, severity)
+
+def to_percentages(build_severities):
+    severities = build_severities[1]
+    total_clients = severities.sum()
+    if total_clients > 0:
+        return (build_severities[0], severities / total_clients)
```

Comment:
No, severities is a Pandas series, which seems to be fine with integer division.